mapbox/vs30/scripts/basin_z_values/gen_sites_in_basin.py

```diff
@@ -0,0 +1,160 @@
+from pathlib import Path
+import time
+import argparse
+
+from numba import jit, njit
+import numba
+import numpy as np
+import pandas as pd
+from Velocity_Model.basins import basin_outlines_dict
+from qcore import geo, formats
+import pygmt
+
+GRID_SPACING = 100
+BASIN_VER = "2.07"
+
+
+@jit(nopython=True)
+def is_inside_postgis(polygon, point):
+    length = len(polygon)
+    intersections = 0
+
+    dx2 = point[0] - polygon[0][0]
+    dy2 = point[1] - polygon[0][1]
+    jj = 1
+
+    while jj < length:
+        dx = dx2
+        dy = dy2
+        dx2 = point[0] - polygon[jj][0]
+        dy2 = point[1] - polygon[jj][1]
+
+        F = (dx - dx2) * dy - dx * (dy - dy2)
+        if 0.0 == F and dx * dx2 <= 0 and dy * dy2 <= 0:
+            return 2
+
+        if (dy >= 0 and dy2 < 0) or (dy2 >= 0 and dy < 0):
+            if F > 0:
+                intersections += 1
+            elif F < 0:
+                intersections -= 1
+
+        jj += 1
+
+    return intersections != 0
+
+
+@njit(parallel=True)
+def is_inside_postgis_parallel(points, polygon):
+    ln = len(points)
+    D = np.empty(ln, dtype=numba.boolean)
+    for i in numba.prange(ln):
+        D[i] = is_inside_postgis(polygon, points[i])
+    return D
+
+
+def get_args():
+
+    parser = argparse.ArgumentParser()
+    arg = parser.add_argument
+    arg(
+        "--outfile",
+        help="Path to the output CSV file",
+        default=Path.cwd() / "sites_in_basin.csv",
+    )
+    arg(
+        "--remove_not_in_basin",
+        help="Remove stations not in basin",
+        action="store_true",
+    )
+    arg("--version", help=f"NZVM version. Default is {BASIN_VER}", default=BASIN_VER)
+    parser = formats.station_file_argparser(parser=parser)
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
+    args = get_args()
+    try:
+        basin_outlines = basin_outlines_dict[arg.version]
+    except KeyError:
+        raise ValueError(f"Basin outlines for version {args.version} not found")
+
+    # Load the boundaries
+    boundaries = {cur_ffp.stem: np.loadtxt(cur_ffp) for cur_ffp in basin_outlines}
+
+    if args.stat_file is None:
+        # remove points in ocean
+        maskvalues = [0, 1, 1, 1, 1]  # ocean / land / lake / island / pond.
+        land_mask = pygmt.grdlandmask(
+            region="NZ",
+            spacing=f"{GRID_SPACING}e/{GRID_SPACING}e",
```

Comment:
should grid spacing maybe be a command line argument? I'd get rid of the `GRID_SPACING` global and replace it with a command line argument defaulting to 100.